### PR TITLE
responsive layout

### PR DESCRIPTION
**Pull request by nhathoang241199**
Responsive layout
Refactor logic code
Responsive upfront pool page
Responsive banner component

### CODE REVIEW

**Review by lhtrung307 — CHANGES_REQUESTED** [review]

File: src/index.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { Suspense } from 'react';
 import ReactDOM from 'react-dom';
+// import 'fonts/Helvetica-Font/Helvetica-Bold.ttf';
```
Comment (lhtrung307) [inline src/index.tsx:3]: Remove comment.

File: src/hooks/useUpfrontPool.ts [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    }
+  };
+
+  const onLeavePool = async (poolPackage: string) => {
+    setSelectedPool(poolPackage);
+    const [account, options] = await getFromAcct(currentAccount);
+    if (api) {
+      const txExecute = api.tx.pool.leave();
+      if (options) {
+        try {
```
Comment (lhtrung307) [inline src/hooks/useUpfrontPool.ts:206]: Duplicate code. Should refactor.

File: src/hooks/useUpfrontPool.ts [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  };
+
+  const onJoinPool = async (poolPackage: string) => {
+    setSelectedPool(poolPackage);
+    const [account, options] = await getFromAcct(currentAccount);
+
+    if (api && account) {
+      const txExecute = api.tx.pool.join({ System: { Upfront: poolPackage } });
+      if (options) {
+        try {
```
Comment (lhtrung307) [inline src/hooks/useUpfrontPool.ts:171]: Duplicate code. Should refactor.

File: src/hooks/useUpfrontPool.ts [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        return {
+          basic: basic.unwrap(),
+          medium: medium.unwrap(),
+          advance: advance.unwrap(),
+        };
+      }
+    }
+  );
+
+  const pools: Array<IPool> = [
```
Comment (lhtrung307) [inline src/hooks/useUpfrontPool.ts:78]: Should not put pool table info inside of this hook.

**Review by lhtrung307 — CHANGES_REQUESTED** [review]

File: src/layouts/DashboardLayout/components/Panel.tsx [hunk 5]
```diff
@@ -1,53 +1,59 @@
...
-        </Heading>
+    <Box w="full">
+      <Box sx={PanelStyled}>
+        <IconButton
+          display={{ base: 'flex', pc: 'none' }}
+          onClick={onOpen}
+          variant="transparents"
+          aria-label="open menu"
+          size="lg"
+          icon={<Icon size={1.5} path={mdiMenu} color={theme.colors.primary} />}
```
[the remark this answers is not in the document]
Comment (nhathoang241199) [reply src/layouts/DashboardLayout/components/Panel.tsx:26]: "primary" dont work in this case.

File: src/themes/theme.tsx [hunk 6]
```diff
@@ -97,292 +96,312 @@ const PanelContent: ComponentStyleConfig = {
...
+);
+
+export const theme = extendTheme({
+  breakpoints: {
+    sm: '320px',
+    md: '768px',
+    lg: '960px',
+    xl: '1200px',
+    '2xl': '1400px',
+    tablet: '740px',
```
[the remark this answers is not in the document]
Comment (nhathoang241199) [reply src/themes/theme.tsx:138]: No, It doesnt. I customized them.

File: src/layouts/DashboardLayout/components/Panel.tsx [hunk 7]
```diff
@@ -57,17 +63,11 @@ export default Panel;
...
-
-const actionStyled = {
-  bg: 'white',
-  color: 'black',
-  w: '56px',
-  h: '56px',
-  borderRadius: '50%',
+  pb: { base: 4, pc: 'none' },
+  p: { base: 'none', pc: 8 },
+  borderBottom: { base: '2px solid #EEF1FF', pc: 'none' },
```
Comment (lhtrung307) [inline src/layouts/DashboardLayout/components/Panel.tsx:72]: Should use color variable here, too.

**Review by lhtrung307 — CHANGES_REQUESTED** [review]

**Review by lhtrung307 — APPROVED** [review]

**Pull request merged**